pkg/http/api.go

```diff
@@ -231,6 +231,34 @@ func SetSecrets(host string, verifyTLS bool, apiKey string, project string, conf
 	return computed, Error{}
 }
 
+// SetSecretNote for specified project and config
+func SetSecretNote(host string, verifyTLS bool, apiKey string, project string, config string, secret string, note string) (models.SecretNote, Error) {
```

Comment:
This might be a bigger refactor so feel free to let me know if it's out of scope.

What do you think of wrapping our HTTP client inside our own struct that would contain the host `localConfig := configuration.LocalConfig(cmd)` at instanciation time. 
This would make it so that we don't have to pass  `host` to `SetSecretNote` It shouldn't be the responsibility of `SetSecretNoet` to know about or worry about the host.

Reply:
This is definitely a larger refactor but I agree there's a better way to be passing this data. At a minimum we should move to using a context, which could contain this and other configuration.

Reply:
If you tackle this in the future or make a linear ticket I'd love to be involved and see how we approach this. I wonder if context is the right spot to put these things.